compiler/rustc_abi/src/map.rs

```diff
@@ -0,0 +1,401 @@
+use std::collections::{BTreeMap, BTreeSet};
+
+use serde_json::Value as JsonValue;
+
+use crate::{AbiFromJsonErr, ArmCall, CanonAbi, ExternAbi, InterruptKind, X86Call};
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct AbiMap {
+    /// a target may support ABIs that are "proper names" for the C ABI, otherwise this is "C"
+    pub c_proper: CanonAbi,
+
+    // required ABIs
+    pub system: CanonAbi,
+    pub system_varargs: CanonAbi,
+    /// ABI used for `extern "rust-cold"`
+    pub rust_cold: CanonAbi,
+
+    // optional architecture-specific ABIs, should always answer a certain way per-arch
+    efiapi: Option<CanonAbi>,
+    win64: Option<CanonAbi>,
+    sysv64: Option<CanonAbi>,
+
+    // optional abstract ABIs, these can be very weird
+    pub stdcall: Option<CanonAbi>,
+    pub fastcall: Option<CanonAbi>,
+    pub thiscall: Option<CanonAbi>,
+    pub vectorcall: Option<CanonAbi>,
+
+    // optional concrete ABIs
+    // arm
+    pub aapcs: bool,
+    pub cmse_nonsecure_entry: bool,
+    // gpu
+    pub gpu_kernel: bool,
+    pub ptx_kernel: bool,
+    // interrupt
+    pub avr_interrupt: bool,
+    pub msp430_interrupt: bool,
+    pub riscv_interrupt: bool,
+    pub x86_interrupt: bool,
+}
+
+// usually the reasonable default, but pay attention to system
+impl Default for AbiMap {
+    fn default() -> AbiMap {
+        let c = CanonAbi::C;
+        let rust_cold = CanonAbi::RustCold;
+        AbiMap {
+            c_proper: c,
+            system: c,
+            system_varargs: c,
+            rust_cold,
+
+            // some mercifully either map or don't
+            // arm
+            aapcs: false,
+            cmse_nonsecure_entry: false,
+            // gpu
+            gpu_kernel: false,
+            ptx_kernel: false,
+            // interrupt
+            avr_interrupt: false,
+            msp430_interrupt: false,
+            riscv_interrupt: false,
+            x86_interrupt: false,
+
+            // x86ish ones lack a mapping by default, but have many
+            efiapi: None,
```

Comment:
efiapi is not an x86ish call conv. It exists on all architectures supported by uefi (currently rustc supports x86, x86_64, arm64, but itanium, arm32, riscv, mips, powerpc64 have either official or unofficial uefi support and thus should support efiapi)

Reply:
oh, but I'm not planning on setting efiapi values for ones that weren't already set before this PR. we apparently have a value set for aarch32 and riscv but not itanium, powerpc64, or mips.